src/main/webapp/app/exercises/programming/shared/code-editor/ace/code-editor-ace.component.ts

```diff
@@ -178,38 +181,39 @@ export class CodeEditorAceComponent implements AfterViewInit, OnChanges, OnDestr
             if ((this.selectedFile && !this.fileSession[this.selectedFile]) || this.fileSession[this.selectedFile].loadingError) {
                 this.loadFile(this.selectedFile);
             } else {
-                this.initEditorAfterFileChange();
+                this.initEditor();
             }
         }
         if (changes.commitState && changes.commitState.currentValue === CommitState.CONFLICT) {
             this.editor.setReadOnly(true);
         } else if (changes.commitState && changes.commitState.previousValue === CommitState.CONFLICT && changes.commitState.currentValue !== CommitState.CONFLICT) {
             this.editor.setReadOnly(false);
         }
+        if (changes.feedback || changes.feedbackSuggestions) {
+            // on changes to these props
+            if (this.editorSession) {
+                // initEditor should already be called, otherwise there is no need to init this anyways
+                this.updateLineWidgets();
```

Comment:
Note for code reviewers: This makes the editor line widgets more robust against changes from "outside"